src/include/sof/audio/component_ext.h

```diff
@@ -216,31 +217,59 @@ struct get_attribute_remote_payload {
 static inline int comp_ipc4_get_attribute_remote(struct comp_dev *dev, uint32_t type,
 						 void *value)
 {
-	struct ipc4_base_module_cfg *base_cfg;
+	struct ipc4_base_module_cfg_ext *basecfg_ext = NULL;
+	struct ipc4_base_module_cfg *base_cfg = NULL;
```

Comment:
you never need both pointers, right? So you could just use one, the only difference is the size, otherwise you'd reuse the same code.

Reply:
two pointers does help with type correctness, in case we ever have to deref. however, I cant see from here if one struct is embedded in the other (then 1 ptr makes sense).

Reply:
no they're not embedded. i will move the free's to the respective switches and remove the inits here